docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+# Practical Example - Birthday Kata
+
+## The Kata
+[Source: Matteo Vaccari](http://matteo.vaccari.name/blog/archives/154)
+
+### Goals
+This kata is normally used to practice TDD, dependency injection, and ports and adapters architectures in an Object Oriented style. Instead, we will tackle the same problem in Unison and explore TDD, dependency rejection, and how pure FP languages naturally encourage ports and adapters like architectures (through seperation of pure and impure functions).
+
```

Comment:
We might need to add a high-level blurb about what the ports and adapters architecture idea is given that it's a central part of the stated goal.